src/sinks/vector/v2.rs

```diff
@@ -0,0 +1,312 @@
+
+// TODO: duplicated in source
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct GrpcTlsConfig {
+    ca_file: PathBuf,
+    crt_file: PathBuf,
+    key_file: PathBuf,
+}
+
+impl GenerateConfig for VectorConfig {
+    fn generate_config() -> toml::Value {
+        toml::Value::try_from(default_config("127.0.0.1:6000")).unwrap()
+    }
+}
+
+fn default_config(address: &str) -> VectorConfig {
+    VectorConfig {
+        address: address.to_owned(),
+        batch: BatchConfig::default(),
+        request: TowerRequestConfig::default(),
+        tls: None,
+    }
+}
+
+lazy_static! {
+    static ref REQUEST_DEFAULTS: TowerRequestConfig = TowerRequestConfig {
+        ..Default::default()
+    };
+}
+
+#[async_trait::async_trait]
+#[typetag::serde(name = "vector")]
+impl SinkConfig for VectorConfig {
+    async fn build(&self, cx: SinkContext) -> crate::Result<(VectorSink, Healthcheck)> {
+        let endpoint = Endpoint::from_shared(self.address.clone())?;
+        let endpoint = match &self.tls {
+            Some(tls) => {
+                let host = get_authority(&self.address)?;
+                let ca = Certificate::from_pem(tokio::fs::read(&tls.ca_file).await?);
+                let crt = tokio::fs::read(&tls.crt_file).await?;
+                let key = tokio::fs::read(&tls.key_file).await?;
+                let identity = Identity::from_pem(crt, key);
+
+                let tls_config = ClientTlsConfig::new()
+                    .identity(identity)
+                    .ca_certificate(ca)
+                    .domain_name(host);
+
+                endpoint.tls_config(tls_config)?
+            }
+            None => endpoint,
+        };
+
+        let client = proto::Client::new(endpoint.connect_lazy()?);
+        let healthcheck = healthcheck(client.clone(), cx.healthcheck.clone());
+
+        let request = self.request.unwrap_with(&REQUEST_DEFAULTS);
+        let batch = BatchSettings::default()
+            .bytes(1300)
+            .events(1000)
+            .timeout(1)
+            .parse_config(self.batch)?;
+
+        let svc = ServiceBuilder::new()
+            .settings(request, VectorGrpcRetryLogic)
+            .service(client);
+
+        let buffer = VecBuffer::new(batch.size);
+        let sink = BatchSink::new(svc, buffer, batch.timeout, cx.acker())
+            .sink_map_err(|error| error!(message = "Fatal Vector GRPC sink error.", %error))
+            .with_flat_map(move |event| stream::iter(encode_event(event)).map(Ok));
+
+        Ok((VectorSink::Sink(Box::new(sink)), Box::pin(healthcheck)))
+    }
+
+    fn input_type(&self) -> DataType {
+        DataType::Any
+    }
+
+    fn sink_type(&self) -> &'static str {
+        "vector_grpc"
+    }
+}
+
+/// Check to see if the remote service accepts new events.
+async fn healthcheck(mut client: Client, options: SinkHealthcheckOptions) -> crate::Result<()> {
+    // TODO: if enabled, allow changing the endpoint through `options.uri`.
```

Comment:
It doesn't seem particularly useful to allow changing this, but I guess since it's a global option that can't be disabled on a per-sink basis, it's best not to make it a no-op.